.github/workflows/pre-commit.yml

```
@@ -0,0 +1,70 @@
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+          
+      - name: Run CloudPosse pre-commit action
+        uses: cloudposse/github-action-pre-commit@v4.0.0
+        with:
+          # Run against all files to ensure comprehensive checks in CI
+          # Local developers run against staged files only
+          extra_args: --all-files
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)